### PR TITLE
UAR-416: Compare and calculate differences for OE data (Bug fixes)

### PR DESCRIPTION
**Pull request by bbell-companieshouse**
### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by AndreKramer7654 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java [hunk 1]
```diff
@@ -5,26 +5,14 @@
 public class EntityNameChange extends Change {
     private static final String CHANGE_NAME = "changeOfEntityName";
 
-    @JsonProperty("corporateBodyName")
-    private String corporateBodyName;
-
-    @JsonProperty("corporateBodyName")
+    @JsonProperty("proposedCorporateBodyName")
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java:8]: Can this property be null?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/PrincipalAddressChange.java [hunk 2]
```diff
@@ -6,26 +6,14 @@
...
     private static final String CHANGE_NAME = "changeOfRoa";
 
-    @JsonProperty("registeredOfficeAddress")
-    private Address registeredOfficeAddress;
-
     @JsonProperty("proposedRegisteredOfficeAddress")
     private Address proposedRegisteredOfficeAddress;
 
-    public PrincipalAddressChange(Address registeredOfficeAddress, Address proposedRegisteredOfficeAddress){
+    public PrincipalAddressChange(Address proposedRegisteredOfficeAddress){
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/PrincipalAddressChange.java:12]: Are any of the properties within the Address object required?

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntityChangeComparatorTest.java [hunk 3]
```diff
@@ -152,99 +157,112 @@ void testCompareCorrespondenceAddressChangeBothInputsNullReturnsNull() {
...
 
         assertNotNull(result);
         assertEquals(CHANGE_COMPANY_IDENTIFICATION, result.getChangeName());
-        assertEquals(updatedLegalForm, result.getProposedLegalForm());
-        assertEquals(TEST_EXISTING_VALUE, result.getProposedGoverningLaw());
-        assertEquals(TEST_EXISTING_VALUE, result.getProposedRegisterLocation());
-        assertEquals(TEST_EXISTING_VALUE, result.getProposedPlaceRegistered());
-        assertEquals(TEST_EXISTING_VALUE, result.getProposedRegistrationNumber());
+        assertEquals(UPDATED_LEGAL_FORM_VALUE, result.getProposedLegalForm());
+        assertEquals(null, result.getProposedGoverningLaw());
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntityChangeComparatorTest.java:190]:
```suggestion
        assertNull(result.getProposedGoverningLaw());
```

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java (hunk 1 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java:8]: Yes, if unchanged so needs a @JsonInclude(NON_NULL)

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java:8]: good catch

**Comment by AndreKramer7654** [comment]
proposedServiceAddress needs a @JOSON_INCLUDE(NOT_NULL) but we can do that in another PR that goes over all the fields.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/PrincipalAddressChange.java (hunk 2 above)
Comment (bmcgreevyCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/PrincipalAddressChange.java:12]: Done

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java (hunk 1 above)
Comment (bmcgreevyCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/EntityNameChange.java:8]: Done.

**Pull request merged**